judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/submission/programming/ContestSubmissionResource.java

```diff
@@ -144,6 +149,12 @@ public ContestSubmissionsResponse getSubmissions(
         userJids = submissions.getPage().stream().map(Submission::getUserJid).collect(Collectors.toSet());
         if (canSupervise) {
             userJids.addAll(contestantStore.getApprovedContestantJids(contestJid));
+            userJids.addAll(supervisorStore
+                    .getSupervisors(contestJid, Optional.empty())
```

Comment:
This will only get the supervisors on page 1. Might need to implement a new `getAllSupervisors()`